lib/task_listener.js

```diff
@@ -212,24 +212,57 @@ export default class TaskListener extends EventEmitter {
     }
   }
 
+  addRunningTask(runningState) {
+    this.runningTasks.push(runningState);
+    this.incrementPending();
+  }
+
+  removeRunningTask(runningState) {
+    let taskIndex = this.runningTasks.findIndex((runningTask) => {
+      return (runningTask.taskId === runningState.taskId) &&
+        (runningTask.runId === runningState.runId)
+    });
```

Comment:
Hrm, I would like to say it was because of some great reason, but really it's just because I didn't think about it. however now thinking about it, there might be a time where something other than someone that holds the reference to runningState wants to remove the task (maybe? no idea).